pandas/_libs/index.pyx

```diff
@@ -40,6 +40,10 @@ cdef inline is_definitely_invalid_key(object val):
             or PyList_Check(val) or hasattr(val, '_data'))
 
 
+def is_definitely_invalid_key(val):
```

Comment:
Easier to just make the original `cpdef bint is_definitely_invalid_key(object val)`?  Plus small perf improvement from adding the bint type.

Reply:
Possible... I have no idea of how ``cdef bint`` compares to ``cdef inline``, which should also bring a small performance improvement.

Reply:
The main motivation would be to avoid needing both a `is_definitely_invalid_key` and `_is_definitely_invalid_key`.  That comes from the `cpdef` part.

bint and inline are not mutually exclusive, so even if you don't use the cpdef suggestion, changing `cdef inline` to `cdef inline bint` is an improvement regardless.

That said, I expect the performance impact to be indistinguishable from noise.

Reply:
(had missed the "p" in ``cpdef``!)